### ISSUE TITLE
`SlevomatCodingStandard.PHP.UselessParentheses` not picking up on `$var = (new ClassName());`

### ISSUE THREAD

**Issue by earthiverse** [slevomat/coding-standard#1686]
Essentially what I wrote in the title.

I added the `SlevomatCodingStandard.PHP.UselessParentheses` sniff to try to pick up on these useless parentheses from a recent code review:

```php
$remoteAddress = (new RemoteAddress());
```

but it never detected it. It did detect a lot of other useless parentheses, though :)

Of course, if I did something like this:

```php
$ip = (new RemoteAddress())->getIpAddress();
```

I expect the sniff not to report as useless (but PHP 8.4 might allow no parentheses around new objects!?)

**Comment by utdrmac** [comment]
I disabled this check because, visually, it makes more sense to me that I'm creating an object. What's the benefit in removing the empty parentheses?

**Comment by kukulich** [comment]
Implemented in https://github.com/slevomat/coding-standard/commit/a7b07d4c21ed3cb15c1e8c137b504fdb8f863ab7
